tests/js/blockchainTest.scrypttest.js

```diff
@@ -0,0 +1,266 @@
+    "merkleRoot": "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60"
+}
+
+
+// it's the secound tx in the block
+let NotLastTxID = '7e0ba1980522125f1f40d19a249ab3ae036001b991776813d25aebe08e8b8a50';
+// merklePath of secound tx in the block
+let merklePathOfNotLastTx = {
+    "blockHash": "0000000000000000091216c46973d82db057a6f9911352892b7769ed517681c3",
+    "branches": [
+        {
+            "hash": "c1d32f28baa27a376ba977f6a8de6ce0a87041157cef0274b20bfda2b0d8df96",
+            "pos": "L"
+        },
+        {
+            "hash": "1e3a5a8946e0caf07006f6c4f76773d7e474d4f240a276844f866bd09820adb3",
+            "pos": "R"
+        }
+    ],
+    "hash": "7e0ba1980522125f1f40d19a249ab3ae036001b991776813d25aebe08e8b8a50",
+    "merkleRoot": "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60"
+}
+
+
+// block hash: 0000000000000000091216c46973d82db057a6f9911352892b7769ed517681c3
+// height: 575191
+let header = {
+    version: 536870912,
+    previousblockhash: "00000000000000000012ce7f4bbce3346ac438ab7fdcb6fa5440db9857856a7f",
+    merkleroot: "95a920b1002bed05379a0d2650bb13eb216138f28ee80172f4cf21048528dc60",
+    time: 1553501874,
+    bits: "180978d5",
+    nonce: 2482491775,
+    difficulty: 116078424449.9863
+}
+
+
+const privateKey = new bsv.PrivateKey.fromRandom('testnet')
+const publicKey = bsv.PublicKey.fromPrivateKey(privateKey)
+
+const tx = newTx();
+const outputAmount = 222222
+
+
+describe('Test sCrypt contract blockchainTest In Javascript', () => {
+    const BlockchainTest = buildContractClass(compileContract('blockchainTest.scrypt'))
+
+    const { BlockHeader, Sibling } = buildTypeClasses(BlockchainTest);
+
+    function buildMerklePath(merklePath) {
+        const array = new Array(32);
+        array.fill(new Sibling({
+            hash: new Sha256("0000000000000000000000000000000000000000000000000000000000000000"),
+            left: 0 //invalid sibling
+        }))
+
+        merklePath.branches.forEach((m, index) => {
+            array[index] = new Sibling({
+                hash: new Sha256(toLittleIndian(m.hash)),
+                left: m.pos === "L" ? 1 : 2
+            });
+        })
+
+        return array;
+
+    }
+
+
+    function toBlockHeader(header) {
+
+        return new BlockHeader({
+            version: new Bytes(uint322bin(header.version)),
+            prevBlockHash: new Sha256(toLittleIndian(header.previousblockhash)),
+            merkleRoot: new Sha256(toLittleIndian(header.merkleroot)),
+            time: new Bytes(uint322bin(header.time)),
+            bits: new Bytes(toLittleIndian(header.bits)),
+            nonce: new Bytes(uint322bin(header.nonce))
+        })
+    }
+
+    before(() => {
+        //Normally, the difficulty of the current network should be used
+        blockchainTest = new BlockchainTest(pdiff2Target(header.difficulty))
+    })
+
+    it('blockchainTest should succeed when using right block header', () => {
```

Comment:
add cases where 

- header validation fails due to low difficulty
- wrong block height